src/glossary.rst

```diff
@@ -4804,6 +4804,16 @@ rustc
 Rust-c (or :dt:`rustc`) is a compiler that implements the Ferrocene Language
 Specification.
 
+.. _fls_Q4MRIo7cWv5K:
+
+safety invariant
+^^^^^^^^^^^^^^^^
+
+:dp:`fls_wRZfAmTmMGTX`
+A :dt:`safety invariant` is an invariant that programs are free to assume to be
+upheld for their execution and when violated may result in :t:`undefined
+behavior`.
```

Comment:
How about something along the lines of

A safety invariant is an invariant that when violated may result in undefined behavior.